tests/zfs-tests/tests/functional/cache/cache.kshlib

```diff
@@ -57,7 +57,7 @@ function display_status
 	((ret |= $?))
 
 	typeset mntpnt=$(get_prop mountpoint $pool)
-	dd if=/dev/random of=$mntpnt/testfile.$$ &
+	dd if=/dev/urandom of=$mntpnt/testfile.$$ &
```

Comment:
This change is triggeting "busy" failures in some `cache` test scripts:

```
Test: /usr/share/zfs/zfs-tests/tests/functional/cache/cache_006_pos (run as root) [00:04] [FAIL]
08:41:00.92 ASSERTION: Exporting and importing pool with cache devices passes.
08:41:01.70 SUCCESS: zpool create testpool /mnt/testdir/disk.cache/a /mnt/testdir/disk.cache/b /mnt/testdir/disk.cache/c cache loop0 loop1
08:41:01.71 SUCCESS: verify_cache_device testpool loop0 ONLINE
08:41:02.36 SUCCESS: zpool export testpool
08:41:03.76 SUCCESS: zpool import -d /mnt/testdir/disk.cache testpool
08:41:05.80 SUCCESS: display_status testpool
08:41:05.81 SUCCESS: verify_cache_device testpool loop1 ONLINE
08:41:05.82 ERROR: zpool destroy testpool exited 1
08:41:05.82 umount: /testpool: target is busy. (In some cases useful info about processes that use the device is found by lsof(8) or fuser(1)) cannot unmount '/testpool': umount failed could not destroy 'testpool': could not unmount datasets
```

We either `s/log_must/log_must_busy/ zpool destroy` or run `sync` just after `kill -9 $dd_pid`: i think the latter is preferred, we shouldn't burden `verify_cache_device()` callers to deal explicitly with an optionally busy fs, it's cleaner to deal with this issue inside the function.

Reply:
@behlendorf i don't know _why_ it's working but _it's working_ indeed, i'll have to RTFM on `wait` (which i think is more lightweight than `sync`, which is good). Thanks.

Reply:
It _should work_ because the problem isn't that there's unwritten dirty data.  This would get written out as part of the unmount regardless.  The problem is that there are still open file handles which are preventing the unmount.  The test need only wait until those child processes exit and the file handles are closed.

@loli10K do you think you're close to have a version of this cleanup ready to merge?  Selfishly I'd love to stop seeing some of these known test failures!

Reply:
This should be almost ready, the only known failure is `inherit_001_pos` which is reproducible and was introduced with the fix for https://github.com/zfsonlinux/zfs/issues/6143, right now i'm working on it.